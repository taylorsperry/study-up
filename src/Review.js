import React from "react";
import CorrectCount from './CorrectCount.js'
import StartOver from './StartOver.js'

const Review = props => {
    return (
      <section className="results-container">
        <article>
          <CorrectCount 
            numCorrect={props.numCorrect}/>
        </article>
          <button disabled={props.reviewLength <= 0} className="result-button" onClick={props.reviewQuizzes}>Review the questions you missed</button>
        <StartOver 
          reset={props.reset}
        />
      </section>
    );
}

export default Review