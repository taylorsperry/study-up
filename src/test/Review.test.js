import React from "react";
import Review from "../Review";
import { shallow } from "enzyme"

describe("Review", () => {
  let wrapper;
  let currentQuiz = {
    "id": 3,
    "question": "If the array 'greeting' has a value of ['Hello', 'World'], and you wanted to return the string 'Hello World', which of the following would you call?",
    "answers": ["greeting.join()", "greeting.concat()", "greeting.join(' ')"],
    "correctAnswer": "greeting.join(' ')",
    "explanation": "Without a separator argument, .join() will return a string with the array’s elements separated by commas; .concat() joins the prototype on which its called with the arguments its passed."
  };
  let reviewQuizzes = [{
    "id": 4,
    "question": "Which of the following array prototype methods does not modify the original array?",
    "answers": [".splice()", ".sort()", ".filter()"],
    "correctAnswer": ".filter()",
    "explanation": ".sort() rearranges the order of the array. .filter() returns a new array based on the criteria it’s passed."
  },
  {
    "id": 5,
    "question": "What happens if you don’t pass two elements for comparison through the .sort() method?",
    "answers": [".sort() will return a copy of the original array", ".sort() will return undefined", ".sort() will rearrange the elements according to their unicode code point values"],
    "correctAnswer": ".sort() will rearrange the elements according to their unicode code point values",
    "explanation": ".sort() can take a compare function, which itself takes two elements to compare."
  }];
  let numCorrect = 7;
  let mockFunc = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
    <Review 
        reviewQuizzes={reviewQuizzes}
        numCorrect={numCorrect}
        reviewLength={reviewQuizzes.length}
        reset={mockFunc}
    />
    )
  });
  
  it("should match snapshot when all data is passed correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
})
