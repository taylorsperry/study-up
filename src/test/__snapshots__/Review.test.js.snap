// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review should match snapshot when all data is passed correctly 1`] = `
<section
  className="results-container"
>
  <article>
    <CorrectCount
      numCorrect={7}
    />
  </article>
  <button
    className="result-button"
    disabled={false}
    onClick={[MockFunction]}
  >
    Review the questions you missed
  </button>
  <StartOver
    reset={[MockFunction]}
  />
</section>
`;
